tests/coverage/closure.coverage

```diff
@@ -173,7 +173,7 @@
    LL|       |
    LL|      1|    let _short_unused_closure_line_break_no_block2 =
    LL|       |        | _unused_arg: u8 |
-   LL|       |            println!(
+   LL|      0|            println!(
```

Comment:
Previously, the span for this `println!(..)` was discarded early, because it covers the whole body span of its surrounding closure. In the new implementation, the span gets truncated to just the `println!` part, so it is retained, giving slightly better instrumentation.

This wasn't an intentional improvement, but I'll take it.